client/src/components/TreeDropdownField/tests/TreeDropdownField-test.js

```diff
@@ -1,684 +1,392 @@
+        treeDropdownField: {
+          ...makeProps().actions.treeDropdownField,
+          addSelectedValues
+        }
+      },
+      visible: [7]
+    })}
+    />
+  );
+  const input = container.querySelector('.treedropdownfield__value-container input');
+  fireEvent.focus(input);
+  fireEvent.keyDown(input, { key: 'ArrowDown', keyCode: 40 });
+  expect(container.querySelector('[id="aria-context"]').textContent).toContain('option page twenty-seven focused, 1 of 2');
+  fireEvent.keyDown(input, { key: 'Enter', keyCode: 13 });
+  expect(onChange).toHaveBeenCalledWith(27);
+  expect(addSelectedValues).toHaveBeenCalledWith(
+    'Form_Test', [{
+      children: [],
+      count: 0,
+      id: 27,
+      title: 'page twenty-seven'
+    }]
+  );
+});
 
-      const newOptions = options.filter(buildTestFilter(field, props.search));
+test('TreeDropdownField handleChange() multi-select should return an array of ids for the selected values', () => {
+  const onChange = jest.fn();
+  const addSelectedValues = jest.fn();
+  const { container } = render(
+    <TreeDropdownField {...makeProps({
+      onChange,
+      actions: {
+        ...makeProps().actions,
+        treeDropdownField: {
+          ...makeProps().actions.treeDropdownField,
+          addSelectedValues
+        }
+      },
+      visible: [7]
+    })}
+    />
+  );
+  const input = container.querySelector('.treedropdownfield__value-container input');
+  fireEvent.focus(input);
+  fireEvent.keyDown(input, { key: 'ArrowDown', keyCode: 40 });
+  expect(container.querySelector('[id="aria-context"]').textContent).toContain('option page twenty-seven focused, 1 of 2');
+  fireEvent.keyDown(input, { key: 'Enter', keyCode: 13 });
+  fireEvent.focus(input);
+  fireEvent.keyDown(input, { key: 'ArrowDown', keyCode: 40 });
+  fireEvent.keyDown(input, { key: 'ArrowDown', keyCode: 40 });
+  expect(container.querySelector('[id="aria-context"]').textContent).toContain('option page fifteen focused, 2 of 2');
+  fireEvent.keyDown(input, { key: 'Enter', keyCode: 13 });
+  expect(onChange).toHaveBeenNthCalledWith(1, 27);
+  expect(onChange).toHaveBeenNthCalledWith(2, 15);
+  expect(addSelectedValues).toHaveBeenNthCalledWith(1, 'Form_Test', [{
+    children: [],
+    count: 0,
+    id: 27,
+    title: 'page twenty-seven'
+  }]);
+  expect(addSelectedValues).toHaveBeenNthCalledWith(2, 'Form_Test', [{
+    children: [],
+    count: 0,
+    id: 15,
+    title: 'page fifteen'
+  }]);
+});
 
-      expect(newOptions).toEqual(options.filter(item => item.value !== 57));
-    });
-  });
+test('TreeDropdownField getVisibleTree() should call findTreeByPath()', () => {
+  const findTreeByPath = jest.fn();
+  render(
+    <TreeDropdownField {...makeProps({
+      tree: {},
+      visible: [1],
+      findTreeByPath
+    })}
+    />
+  );
+  expect(findTreeByPath).toHaveBeenCalledWith({}, [1]);
+});
 
-  describe('getPath()', () => {
-    it('should return a breadcrumb path separated by /', () => {
```

Comment:
This test is now covered by `TreeDropdownField getBreadcrumb() should traverse the path given and return the relevant nodes in the tree`